packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -397,8 +397,8 @@ export class AuthTransformer extends TransformerAuthBase implements TransformerA
       const subscriptionRoles = acm
         .getRolesPerOperation('listen')
         .map(role => this.roleMap.get(role)!)
-        // for subscriptions we only use static rules or owner rule
-        .filter(roleDef => roleDef.strategy === 'owner' || roleDef.static);
+        // for subscriptions we only use static rules or owner rule where the field is not a list
+        .filter(roleDef => (roleDef.strategy === 'owner' && !fieldIsList(def.fields ?? [], roleDef.entity!)) || roleDef.static);
       subscriptionFieldNames.forEach(subscription => {
         this.protectSubscriptionResolver(context, subscription.typeName, subscription.fieldName, subscriptionRoles);
```

Comment:
Can you please double-check if the other changes to this file in [the original PR](https://github.com/aws-amplify/amplify-category-api/pull/551/files#diff-0a38964c5525097a10b63bbbbea2f24278afce76a68e59353eaeb7d805372314R955) need to be reverted?

Reply:
No need to revert those changes, they don't impact the functionality.